internal/graph/memoryWithDelete_test.go

```diff
@@ -0,0 +1,259 @@
+/*
+Copyright The ORAS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package graph
+
+import (
+	"context"
+	"sync"
+	"testing"
+
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"oras.land/oras-go/v2/internal/descriptor"
+)
+
+var (
+	A    = ocispec.Descriptor{MediaType: "A"}
+	AKey = descriptor.FromOCI(A)
+	B    = ocispec.Descriptor{MediaType: "B"}
+	BKey = descriptor.FromOCI(B)
+	C    = ocispec.Descriptor{MediaType: "C"}
+	CKey = descriptor.FromOCI(C)
+	D    = ocispec.Descriptor{MediaType: "D"}
+	DKey = descriptor.FromOCI(D)
+)
+
+// 条件：
+// 1. 只要node被index（作为函数的第二个输入），node就在predecessors，successors，indexed中有entry
```

Comment:
Can you please translate the comments to English? =)

Reply:
I will remove it or put it formally in English in later versions. Currently this is a draft PR, and I wrote this comment to remind myself the designs.